src/Controller/FilesController.php

```diff
@@ -78,16 +81,28 @@ public function fileAction(Request $request, string $file): BinaryFileResponse
         $allowLogin = false;
         $allowAccess = false;
 
+        // Get the current user
+        $user = $this->security->getUser();
+
+        // Check if the current user can access their home directory
+        $canAccessHomeDir = $user instanceof FrontendUser && !empty($user->homeDir) && $user->accessHomeDir;
+
         do {
             // Only check for folders and when member groups have been set
-            if ('folder' === $filesModel->type && null !== $filesModel->groups) {
+            // or access to member home directory
+            if ('folder' === $filesModel->type
+                && (null !== $filesModel->groups
+                    || ($canAccessHomeDir))) {
```

Comment:
```suggestion
            if ('folder' === $filesModel->type && (null !== $filesModel->groups || $canAccessHomeDir)) {
```

Reply:
Wouldn't it be simpler to ask before the loop whether the user is logged in? the method isVisibleElement does that too, but in case of doubt it also does so for each loop pass.

Without a login, the check is not necessary overall.